include/z64scene.h

```diff
@@ -469,6 +469,14 @@ typedef enum {
 #define SCENE_CAM_TYPE_FIXED_MARKET 0x40 // Camera exhibits fixed behaviors and delays textboxes by a small amount before they start to appear
 #define SCENE_CAM_TYPE_SHOOTING_GALLERY 0x50 // Unreferenced in code, and used only by the main layer of the shooting gallery scene
 
+// navi hints
+// TODO: make ZAPD use this enum for `SCENE_CMD_SPECIAL_FILES`
+typedef enum {
+    NAVI_HINTS_NONE,
+    NAVI_HINTS_OVERWORLD,
+    NAVI_HINTS_DUNGEON
+} NaviHintFileId;
```

Comment:
Having an enum for this feels unnecessary and somewhat incorrect since it's really just an index in `sNaviHintFiles`, tho it's 1-indexed, as indicated by the - 1 in `Scene_CommandSpecialFiles`, but I guess it's debatable

Reply:
the -1 is precisely my reasoning for doing this, you cant claim that the number provided to the scene command is an index when its not the value that actually indexes the list. its more of an id where 0 is none

Reply:
It's 1-indexed but it's still an index since the only thing it's used for is to index into the array

Reply:
I dont agree but fair enough
I change it back to calling it an index in the code, but I think the enum is still nicer and easier to understand as dragorn has shown above

Reply:
The actor table has enums, the object table has enums, the scene table has enums, I assume the entrance table has enums although with that who knows. I don't think being primarily an index into an array, albeit a small and little-used one, precludes it from having an enum.

Reply:
If that weighs anything, all modding tools use a dropdown list for this: none/overworld/dungeon

Reply:
To be clear I'm not saying it being an index prevents it from being associated with an enum, but I think this is a case where it's unnecessary as the values are never referenced in code and having names for them doesn't serve much purpose (and thus might not be worth having)

Reply:
alright. but yeah, even if they arent referenced in code (other than 0/none), it makes the usage in scene files alot clearer so it is worth it in my opnion